tensorflow/contrib/framework/python/ops/variables_test.py

```diff
@@ -810,5 +810,27 @@ def testInitFromCheckpointWithScopes(self):
       self.assertAllEqual(init_value0, var0.eval())
       self.assertAllEqual(init_value1, var1.eval())
 
+class ZeroInitializerOpTest(tf.test.TestCase):
+
+  def _testZeroInitializer(self, shape, dtype):
+    var0 = tf.Variable(tf.zeros(shape, dtype=dtype))
+    var1 = tf.Variable(tf.ones(shape, dtype=dtype))
+    var0_zero = tf.contrib.framework.zero_initializer(var0)
+    var1_zero = tf.contrib.framework.zero_initializer(var1)
```

Comment:
My suggestion was that you make an op which takes any constant float value (0.0 by default) and initializes the variable to have that value repeated in all entries. Then you can replace also things which are not zero initialized (like your ones-initialized variable, which weirdly gets initialized with zero by your zero_initializer op).

I'd also like it if the syntax was more like var0 = tf.Variable(tf.zero_initializer(shape, dtype)) to make it work seamlessly with the existing infrastructure such as initialize_all_variables.